scripts/west_commands/zephyr_ext_common.py

```diff
@@ -16,6 +16,13 @@
 
 from runners.core import RunnerConfig
 
+# This relies on this file being zephyr/scripts/foo/bar.py.
+# If you move this file, you'll break it, so be careful.
+THIS_ZEPHYR = Path(__file__).parent.parent.parent
+ZEPHYR_BASE = Path(os.environ.get('ZEPHYR_BASE', THIS_ZEPHYR))
```

Comment:
Sorry, I misspoke in my earlier comment.

This is only really a fix if west does *not* set ZEPHYR_BASE for us when we run extensions. Which I set up when I was testing this locally to see what would break, if anything.

So feel free to NAK these changes, but I assume the future we are working towards is "west doesn't know anything about zephyr", so I'd like to get these changes in along with this PR to be ready for that along with the rest of these improvements.

Reply:
Thanks for the fixes.
Let's so how much work it is to get rid of the remaining Zephyr base.

Reply:
I'm able to do pristine build when removing those 4 lines :)
Going to work further on this later.

Reply:
> I'm able to do pristine build when removing those 4 lines :)

Right: 

> This is only really a fix if west does _not_ set ZEPHYR_BASE for us when we run extensions. Which I set up when I was testing this locally to see what would break, if anything.

Reply:
Another thing to consider, is that the `ZEPHYR_BASE` determined this way could be different from the `ZEPHYR_BASE` used when building the project.

For a command, such as pristine, this may not matter much, as that cmake file has not changed in >2 years.
But for other commands, things could behave different if using a CMake file found this way, instead of the CMake file set on first CMake invocation

So this is just one more reason to clean up this.

Reply:
@mbolivar No need for Zephyr base setting before executing CMake.
Pristine now obtain the cached Zephyr base from CMakeCache.txt, which is the correct way, as it ensures pristine will always use the same ZEPHYR_BASE that was used on first CMake invocation.

`west boards` will use the `find_package(Zephyr)` inside the script to ensure that Zephyr base is determined in same way as we do for that samples.

Only a single note on this, which is that we still need to know the location of the CMake script, as there is no build folder where this information can be obtained. But that is similar to the `script` folder issue, and no longer a Zephyr base environment setting. 

@mbolivar and @carlescufi I also changed `BOARD_ROOT_SPACE_SEPARATED` to simply `BOARD_ROOT`, in boards.cmake and also adds `ZEPHYR_BASE` as board root.
This is done in order to be consistent with existing Zephyr docs when building, and boilerplate code.
See those two comments:
https://github.com/zephyrproject-rtos/zephyr/pull/23054/commits/88b120445945decb572ecf521a0ae14f4c95f0db#r390341804
https://github.com/zephyrproject-rtos/zephyr/pull/23054/commits/88b120445945decb572ecf521a0ae14f4c95f0db#r390343152